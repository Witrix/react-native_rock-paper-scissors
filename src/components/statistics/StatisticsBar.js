import React, { Component } from 'react';
import {
    View,
    Text, StyleSheet,
} from 'react-native';
import AppColors            from "../../themes/colors";

export default class StatisticsBar extends Component {

    constructor(props) {
        super(props);
        let percent = 0.01;
        if (props.percent && typeof props.percent === 'number' && props.percent >= 0) percent = props.percent;
        this.state = {
            percent: percent,
            color: props.color,
        }
    }

    componentWillReceiveProps(nextProps) {
        let percent = 0.01;
        if (nextProps.percent && typeof nextProps.percent === 'number' && nextProps.percent >= 0) percent = nextProps.percent;
        this.setState({
            percent: percent,
            color: nextProps.color,
        })
    }

    render() {
        return (
            <View style={{flex: 1, flexDirection: 'row', margin: 1}}>
                <View style={{flex: this.state.percent, backgroundColor: this.state.color || AppColors.brand.primary, height: 10}}/>
                <View style={{flex: 1 - this.state.percent, height: 10}}/>
            </View>
        );
    }
}
