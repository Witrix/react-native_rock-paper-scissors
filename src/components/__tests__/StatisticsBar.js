import 'react-native';
import React                    from 'react';
import StatisticsBar            from '../statistics/StatisticsBar';
import renderer                 from 'react-test-renderer';

it('Render correctly', () => {
  const tree = renderer.create(
    <StatisticsBar />
  );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: valid props', () => {
    const tree = renderer.create(
        <StatisticsBar percent={0.5} color={'red'}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: invalid props', () => {
    const tree = renderer.create(
        <StatisticsBar percent={"abc"} color={-1}/>
    );
    expect(tree).toMatchSnapshot();
});