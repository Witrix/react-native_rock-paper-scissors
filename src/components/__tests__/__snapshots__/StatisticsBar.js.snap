// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "margin": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#29B6F6",
        "flex": 0.01,
        "height": 10,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 0.99,
        "height": 10,
      }
    }
  />
</View>
`;

exports[`Render correctly: invalid props 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "margin": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": -1,
        "flex": 0.01,
        "height": 10,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 0.99,
        "height": 10,
      }
    }
  />
</View>
`;

exports[`Render correctly: valid props 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "margin": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "flex": 0.5,
        "height": 10,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 0.5,
        "height": 10,
      }
    }
  />
</View>
`;
